### ISSUE TITLE
Failed to copy manifest with 0 size config

### ISSUE THREAD

**Issue by northtyphoon** [oras-project/oras-go#129]
I tried to copy a manifest from registry `binducentralus.azurecr.io/samples/artifact:1.0`. This manifest has a config whose size is 0. 

```
{
  "schemaVersion": 2,
  "config": {
    "mediaType": "application/vnd.unknown.config.v1+json",
    "digest": "sha256:e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855",
    "size": 0
  },
  "layers": [
    {
      "mediaType": "application/vnd.unknown.layer.v1+txt",
      "digest": "sha256:6291bbe45f529c52451285872d5fce81296aafac2190634d6f2e17c08c2e3dcb",
      "size": 44,
      "annotations": {
        "org.opencontainers.image.title": "artifact.txt"
      }
    }
  ]
}
```

The following code failed as 400 bad request. 

```
src, err := remote.NewRepository("binducentralus.azurecr.io/samples/artifact:1.0")
dest := file.New("temp")
desc, err := oras.Copy(ctx, src, src.Reference.Reference, dest, "")
```

The problem was the invalid range header value (`bytes=0--1`) in the download request due to https://github.com/oras-project/oras-go/blob/c34895e747dcd988adb3cc959302b5e94117195b/registry/remote/repository.go#L447.

**Comment by wangxiaoxuan273** [comment]
Please assign this issue to me, thanks

**Comment by shizhMSFT** [comment]
Resolved by #145
